Components/IconTypes/IconType_customvalue/customvalue.lua

```diff
@@ -0,0 +1,108 @@
+﻿local TMW = TMW
+if not TMW then return end
+
+local L = TMW.L
+
+local print = TMW.print
+
+
+local Type = TMW.Classes.IconType:New("customvalue")
+
+Type.name = L["ICONMENU_CUSTOMVALUE"]
+Type.desc = L["ICONMENU_CUSTOMVALUE_DESC"]
+Type.menuIcon = "Interface/Icons/inv_misc_punchcards_white"
+
+Type.hasNoGCD = true
+Type.canControlGroup = true
+Type.menuSpaceBefore = true
+Type.barIsTimer = false
+Type.menuSpaceBefore = true
+
+
+
+local STATE_SUCCEED = TMW.CONST.STATE.DEFAULT_SHOW
+local STATE_FAIL = TMW.CONST.STATE.DEFAULT_HIDE
+
+Type:SetAllowanceForView("icon", false)
+Type:SetModuleAllowance("IconModule_TimerBar_Overlay", false)
+Type:SetModuleAllowance("IconModule_CooldownSweep", false)
+Type:UsesAttributes("texture")
+
+Type:RegisterIconDefaults{
+	func = nil
+}
+
```

Comment:
There is no icon setting named `func` that is being configured, so this is not needed. `func` is just a transient property stored on the Icon object - these do not need to be declared.

However, a new setting for storing the user's Lua code (as I mentioned in other comments) would necessitate a default being declared here.